### ISSUE TITLE
Why fedora?

### ISSUE THREAD

**Issue by wolfeidau** [desdrury/k8sdc#4]
Just been running through the setup and wanted to ask why you used fedora?

Cheers

**Comment by desdrury** [comment]
Really just habit.  I needed Systemd but RHEL and CentOS did not have it when I started playing around with Docker and then Kubernetes.  I wanted to use a Red Hat distribution of some type as this was what I mostly encountered in my work environments.  I did use CoreOS quite sometime ago but found the lack of a package manager frustrating when trying install tools to debug issues.  Do you think there is a more appropriate choice?

**Comment by wolfeidau** [comment]
Yeah I think Centos 7 is the way to go if you want to support customers using Redhat. There are a couple of great centos based vagrant boxes like https://atlas.hashicorp.com/puppetlabs/boxes/centos-7.2-64-nocm which support vbox/vmware.

Also was there a reason you don't use the yum repos for docker?

Cheers

**Comment by desdrury** [comment]
I'd be happy to switch to Centos 7.  Did yo want to add it in your fork?  I'll purchase copy of the Vagrant VMWare Fusion plugin so I can also test on both vbox/vmware.  

The reason I didn't use the yum repos for Docker is that I wanted to decouple the base Docker / Flannel / Etcd / Kubernetes binaries from a particular distributions packager manager.  This allows the latest version to be installed when it is released as opposed to when a package manager has it available.  It also has the added benefit of being able to download these binaries to the "fileserver" once and then each Kubernetes node can pull them from the "fileserver" much more quickly.  This also allows for and organisation to nominate just one server with access to the internet.

**Comment by wolfeidau** [comment]
Yeah the fedora image was a total pain in the butt because of the interface names were different between vbox and vmware. Vbox had eth0/eth1 and vmware ens32 / ens33, this is suppose to be fixed in a more recent version of fedora/vagrant I believe..

Because you need the bloody interface name to setup stuff inside ansible i didn't see a simple way to decouple this in the existing ansible roles so it kinda stalled, hence no pull request :)

Yeah interesting, I like to use YUM repos for this on production boxes as the packages are then signed, even if they are just created with basic fpm script :)

I will have a look at the CentOS thing over Christmas / New years by pulling it apart and breaking some things.

Enjoy your break!

**Comment by desdrury** [comment]
You enjoy to your break too :)  And a happy Xmas / New Year!

**Comment by wolfeidau** [comment]
Just an update, I started building a version of this that works with centos 7.

I need to sort out some issues around the NFS setup, at some point we should do coffee and discuss the reasoning around this setup in more detail :)

Hope your enjoying your break.

**Comment by desdrury** [comment]
Thanks for the update :)

NFS was a problem with Vagrant. Some obscure bug. So I had to make some changes to the playbook.   I also created a utility to fix things up when restarting the k8sdc cluster. 

https://k8sdc.github.io/reference/utilities/#vagrant-restart-fix

I'm going to be out of mobile coverage for the next 3 days but a catch up for coffee sounds great :)  Maybe sometime the week beginning 25th Jan?
